src/System.IO.FileSystem/tests/FileSystemTest.cs

```diff
@@ -35,6 +35,13 @@ public string GetTestFilePath([CallerMemberName]string fileName = null, [CallerL
             return Path.Combine(TestDirectory, String.Format("{0}_{1}", fileName ?? "testFile", lineNumber));
         }
 
+        // Generates a test file path to use that is unique name per test case / call with an additional
+        // variable to specify a trailing identifier
+        public string GetTestFilePath(int index, [CallerMemberName]string fileName = null, [CallerLineNumber] int lineNumber = 0)
```

Comment:
I just noticed that the parameter name is fileName on all these rather that memberName. It should probably be the latter. At some point we should also consolidate such base classes we have in a few different test projects and put it in common.